app/scripts/lib/setupSentry.js

```diff
@@ -624,3 +646,27 @@ function getEventType(event) {
 
   return 'Event';
 }
+
+// Add this function to directly get the metaMetricsId from various sources
+async function getMetaMetricsId() {
+  try {
+    const state = getState();
+    let metaMetricsId =
+      state?.state?.MetaMetricsController?.metaMetricsId ||
+      state?.state?.metamask?.metaMetricsId;
+
+    const persistedState = await globalThis.stateHooks.getPersistedState();
```

Comment:
Q: don't you want to stop looking for a value if you found one? the first from state is going to be overwritten by persisted state.
Why not return once you have a correct one?
Are the state and persistedState exclusive so there's never this risk of overwriting?

Reply:
Good spot ! Adjust solution to use uuidV4 generated per session as discussed in slack thread
- 7b1f71740541cb39f1b383f90887e9c80ee70fad
- 7c2bf713b5ff40c636650b4a9b6695dfdd642553